### PR TITLE
[XamlC] Add netstandard2.0 to Xamarin.Forms.Build.Tasks

### PR DESCRIPTION
**Pull request by WonyoungChoi**
### Description of Change ###

This is a modification to build the Xamarin.Forms application including xaml files using dotnet-cli (>=.NETCore 2.0).

Previously, when trying to build an app with dotnet-cli, the following error occurred.

> Could not load file or assembly 'Microsoft.Build.Utilities.v4.0, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'

The reason of this error is that Xamarin.Forms.Build.Tasks.dll was built with NET4x target framework.
For dotnet-cli, it must be built with NETStandard or NETCoreApp.

This PR has following changes.
- Add Xamarin.Forms.Build.Tasks.Core project for netstandard2.0 build
- Fix some code of Xamarin.Forms.Build.Tasks to build with netstandard2.0
- Add netstandard2.0 dlls to Xamarin.Forms nuget package

Related PR: https://github.com/xamarin/Xamarin.Forms/pull/868

### Bugs Fixed ###

https://bugzilla.xamarin.com/show_bug.cgi?id=60705
see dotnet/cli#5457 (comment) for a repro

### API Changes ###

N/A

### Behavioral Changes ###

Xamarin.Forms.Build.Tasks.dll, built with netstandard2.0, is used when use dotnet-cli.

### PR Checklist ###

- [ ] Has tests (if omitted, state reason in description)
- [X] Rebased on top of master at time of PR
- [X] Changes adhere to coding standard
- [X] Consolidate commits as makes sense

### CODE REVIEW

**Comment by StephaneDelcroix** [comment]
Hey @WonyoungChoi 

Thanks for your first PR 🎉 

It almost comes at the right time, just a few days/weeks too soon, as, for the time being, we still need to be able to build on toolchain that does not support netstandard2. But we're busy fixing that.

At first glance, this looks right, but I have to:
 - test it
 - review the changes in cecil 0.10.0-beta7 (b6 still had issue with assemblyresolution for netstandard)

and after that I'll come back to you

**Comment by WonyoungChoi** [comment]
I've rebased on latest master including https://github.com/xamarin/Xamarin.Forms/pull/1306

**Comment by WonyoungChoi** [comment]
@StephaneDelcroix Ping. Could you review this PR again?

**Comment by StephaneDelcroix** [comment]
I'm getting 

```
Project Xamarin.Forms.Build.Tasks is not compatible with netstandard2.0 (.NETStandard,Version=v2.0). Project Xamarin.Forms.Build.Tasks supports: net47 (.NETFramework,Version=v4.7)
```
while restoring packages

**Comment by WonyoungChoi** [comment]
@StephaneDelcroix I can't see any errors while restoring in my environment. Xamarin.Forms.Build.Tasks.csproj defines netstandard2.0 and net47 both. Try removing obj, bin directories of Xamarin.Forms.Build.Tasks manually because the project-system of this project was changed. 
If you get same errors, please leave a more detailed error message.

**Comment by StephaneDelcroix** [comment]
@WonyoungChoi this happens on VSforMac

**Comment by WonyoungChoi** [comment]
@StephaneDelcroix Unfortunately, VSforMac doesn't support multiple target frameworks in single csproj yet. 😭   https://bugzilla.xamarin.com/show_bug.cgi?id=56809
I'll separate the .csproj for netstandard version.

**Comment by StephaneDelcroix** [comment]
I'd prefer not having to separate the projects. I'm checking about the status of that bug in VSfM

**Comment by WonyoungChoi** [comment]
@StephaneDelcroix This project separation is workaround. The target milestone of that bug is 15.6 but we can't wait until then. After the bug has been fixed, it may be better to rearrange the project.

File: .nuspec/Xamarin.Forms.targets [hunk 1]
```diff
@@ -18,7 +26,7 @@
 	<PropertyGroup>
 		<_XFTargetsImportedAgain Condition="'$(_XFTargetsImported)'=='True'">True</_XFTargetsImportedAgain>
 		<_XFTargetsImported>True</_XFTargetsImported>
-		<IsXBuild Condition="'$(MSBuildRuntimeVersion)' == ''">true</IsXBuild>
+		<IsXBuild Condition="'$(MSBuildRuntimeVersion)' == '' and '$(MSBuildRuntimeType)' != 'Core'">true</IsXBuild>
```
Comment (StephaneDelcroix) [inline .nuspec/Xamarin.Forms.targets]: if the left part of the comparison is `true`, so is the second. you might mean `or`

**Comment by WonyoungChoi** [comment]
I changed the comparion variable to determine XBuild. 
Following is my test result about MSBuildRuntimeType and MSBuildRuntimeVersion.

(runtime) : (MSBuildRuntimeType) : (MSBuildRuntimeVersion)
xbuild : 'empty' : 'empty'
msbuild : 'Full' : '4.0.30319'
msbuild (mono) : "Mono" : "4.0.30319"
dotnet msbuild : "Core" : 'empty'

**Pull request merged**